CMakeLists.txt

```diff
@@ -7,6 +7,7 @@ cmake_minimum_required(VERSION 3.15)
 
 cmake_policy(SET CMP0079 NEW)
 cmake_policy(SET CMP0091 NEW)
+cmake_policy(SET CMP0118 NEW)
```

Comment:
```
Policy "CMP0118" is not known to this version of CMake.
```

CMake 3.19.20122902-MSVC_2.

Reply:
Simply removing the line works. Why is it required?

Reply:
How silly is that. I was thinking that cmake would ignore polcy settings for unknown ones for backwards compatibility :roll_eyes: 

Not setting this results in a cmake warning with cmake >= 3.20. The policy is (`cmake --help-policy CMP0118`)

CMP0118
-------

The ``GENERATED`` source file property is now visible in all directories.

Whether or not a source file is generated is an all-or-nothing global
property of the source.  Consequently, the associated ``GENERATED``
property is now visible from any directory scope, not only from the scope
for which it was set.

Additionally, the ``GENERATED`` property may now be set only to boolean
values, and may not be turned off once turned on.

The ``OLD`` behavior of this policy is to only allow ``GENERATED`` to be
visible from the directory scope for which it was set.  The ``NEW``
behavior on the other hand allows it to be visible from any scope.

This policy was introduced in CMake version 3.20.  Use the
``cmake_policy()`` command to set it to ``OLD`` or ``NEW`` explicitly.
Unlike many policies, CMake version 3.20.1 does *not* warn
when this policy is not set and simply uses ``OLD`` behavior with regard
to visibility of the ``GENERATED`` property.  However, CMake does warn
about setting the ``GENERATED`` property to a non-boolean value.

----

So I would say this does not change anything for us atm, but in order to get rid of the warning I set the policy. I guess this means that we'll have to only set the policy if the used cmake version is high enough...

Reply:
That's what I thought.